jerry-core/jerryscript.h

```diff
@@ -166,6 +166,29 @@ typedef void (*jerry_object_free_callback_t) (const uintptr_t native_p);
 typedef bool (*jerry_object_property_foreach_t) (const jerry_value_t property_name,
                                                  const jerry_value_t property_value,
                                                  void *user_data_p);
+/**
+ * Type information of a native handle.
+ */
+typedef struct
+{
+  jerry_object_free_callback_t free_cb; /**< the free callback of the native handle */
+} jerry_native_handle_info_t;
+
+/**
+ * The name of the jerry_native_handle_info_t struct.
+ */
+#define JERRY_NATIVE_HANDLE_INFO_FOR_CTYPE(c_type) _jerry_native_handle_info_##c_type
+
+/**
+ * Define a native handle typeinfo based on the C type and free callback.
+ */
+#define JERRY_DEFINE_NATIVE_HANDLE_INFO(c_type, native_free_cb) \
+  static const jerry_native_handle_info_t JERRY_NATIVE_HANDLE_INFO_FOR_CTYPE (c_type) = \
+  { \
+    .free_cb = (jerry_object_free_callback_t) native_free_cb \
+  }
```

Comment:
My idea was to provide a macro to make it easy to create handle infos per C (struct) type:

```
// my_binding.c

typedef struct {
  int native_obj_state;
  ...
} MyNativeObj;

static void free_my_native_obj(uintptr_t handle);

JERRY_DEFINE_NATIVE_HANDLE_INFO(MyNativeObj, free_my_native_obj);
// ^-- creates `static const jerry_native_handle_info_t _jerry_native_handle_info_MyNativeObj`

... {
  MyNativeObj *native_obj = malloc(...);
  jerry_set_object_native_handle(jerry_value, (uintptr_t)native_obj,
                       &JERRY_NATIVE_HANDLE_INFO_FOR_CTYPE(MyNativeObj));
}

```

I think we need to add docstrings for the arguments and an example as well (probably a simple one in the docstrings?).

> How can I extend jerry_native_handle_info_t with other read-only members?

I did not take additional, custom struct members into account. Do you think this is so common that it's worth complicating the macro further to accommodate for this?
You can create a new struct type that wraps `jerry_native_handle_info_t` and add your fields there, i.e.: